lib/components/molecules/Carousel/Carousel.js

```diff
@@ -0,0 +1,398 @@
+  );
+};
+
+/**
+ * Carousel Main Element.
+ *
+ * @param {Element} children
+ *   Wraps all tha carousel elements.
+ *
+ * @return Carousel.
+ */
+const Carousel = ({
+  children,
+  className,
+  nav,
+  title,
+  autoplay,
+  slideTransitionTime,
+}: Props): Node => {
+  const [slideIndex, setSlideIndex] = useState(1);
+  const [slides, setSlidesContent] = useState([]);
+  const [slideCount, setSlidesCount] = useState(1);
+  const [navigation, setNavigation] = useState(nav);
+  const [isAnimationRunning, setAnimationRunning] = useState(autoplay);
+
+  const setData = data => setSlidesContent([...slides, ...data]);
+
+  useInterval(
+    () => {
+      const currentIndex = slideIndex !== slideCount ? slideIndex : 1;
+      setSlideIndex(currentIndex);
+    },
+    isAnimationRunning ? slideTransitionTime : null
+  );
+
+  const value = {
+    slides,
+    setData,
+    slideIndex,
+    setSlideIndex,
+    setSlidesCount,
+    slideCount,
+    setNavigation,
+    setAnimationRunning,
+    isAnimationRunning,
+  };
+
+  return (
+    <section className={classnames('carousel-wrapper', className)} aria-label={title}>
+      <CarouselContext.Provider value={value}>
+        {navigation && nextKey(setSlideIndex, slideIndex, slideCount, null)}
+        {children}
+        {navigation && prevKey(setSlideIndex, slideIndex, slideCount, null)}
+        {ariaLiveUpdate(slideIndex, slideCount)}
+      </CarouselContext.Provider>
+    </section>
+  );
+};
+
+/**
+ * Slide Wrapper.
+ *
+ * @param {Array} children
+ *   Wraps all the slides.
+ *
+ * @return All slides in single wrapper.
+ */
+Carousel.SlideWrapper = ({ children, className }: Props): Node => {
+  const slideContent = useContext(CarouselContext);
+  let currentAnimationStatus = true;
+
+  if (slideContent.slides.length === 0) {
+    const results = slideWithIndexKey(children);
+    slideContent.setData(results);
+    slideContent.setSlidesCount(results.length);
+  }
+
+  const handleMouseEnter = () => {
+    if (slideContent.isAnimationRunning) {
+      currentAnimationStatus = true;
+      slideContent.setAnimationRunning(false);
+    } else {
+      currentAnimationStatus = false;
+    }
+  };
+
+  const handleMouseLeave = () => {
+    if (currentAnimationStatus) {
+      currentAnimationStatus = false;
+      slideContent.setAnimationRunning(true);
+    }
+
+    return null;
+  };
+
+  return (
+    <div
+      className={classnames('carousel-section', className)}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      {slideContent.slides}
+    </div>
+  );
+};
+
+/**
+ * Slide Container.
+ *
+ * @param {Element} children
+ *   Will wraps the slide content.
+ *
+ * @return Single slide with wrapper.
+ */
+Carousel.Slide = ({ children, id, className }: Props): Node => {
+  const SlideStatus = useContext(CarouselContext);
+
+  return (
+    <div
+      className={classnames(
+        'slide',
+        { active: id === `slide_${SlideStatus.slideIndex}` },
```

Comment:
There is no need for such comparison use the ActiveSlide State to make this decision.